src/packet_distribution.cpp

```diff
@@ -478,37 +478,18 @@ void AOApplication::server_packet_received(AOPacket *p_packet)
 
     bool musics_time = false;
     area_count = 0;
-    bool legacy_system = false;
     int element_ahead = 0;
     for (int n_element = 0; n_element < f_contents.size(); ++n_element) {
       element_ahead = n_element + 1;
+      // Here we check whether we are using the old music list or the new one.
+      // We do that by checking if there is a '==' in the current element and a
+      // song in the next or if we are currently on a song
       if (!musics_time && f_contents.at(n_element).startsWith("==") &&
-          (f_contents.at(element_ahead).endsWith(".wav") ||
-           f_contents.at(element_ahead).endsWith(".mp3") ||
-           f_contents.at(element_ahead).endsWith(".mp4") ||
-           f_contents.at(element_ahead).endsWith(".ogg") ||
-           f_contents.at(element_ahead).endsWith(".opus"))) {
-        legacy_system = true;
+          is_music_track(f_contents.at(element_ahead))) {
+        musics_time = true;

```

Comment:
Will crash if the only entry on the music list is an empty category, though that is not a regression you caused.